src/compiler/generate-common-tests.xsl

```diff
@@ -721,19 +734,36 @@
          else QName($thisvar/@namespace-uri, $thisvar/@name)
          "/>
       <xsl:variable name="distinctqnames" as="xs:QName*" select="distinct-values($qnames)"/>
-      <!-- Return distinct strings, except that any unprefixed name with nonempty URI
-         needs Q{} notation to record the namespace URI. -->
-      <xsl:sequence select="for $thisqname in $distinctqnames return
-         if ( empty(prefix-from-QName($thisqname)) and (string-length(namespace-uri-from-QName($thisqname)) gt 0) )
-         then concat('Q{',namespace-uri-from-QName($thisqname),'}',local-name-from-QName($thisqname))
-         else string($thisqname)
-         "/>
+      <!-- Return distinctly named <x:var> elements. Any unprefixed name with nonempty URI
+         or any prefixed name that is not uniquely bound in the set of variables
+         uses Q{} notation to record the namespace URI. -->
+      <xsl:for-each select="$distinctqnames">
+         <xsl:variable name="thisqname" select="."/>
+         <xsl:variable name="thisprefix" select="prefix-from-QName($thisqname)"/>
+         <x:var>
```

Comment:
This function is namespace-sensitive. Then, isn't it safer to construct the `x:var` element using `<xsl:element>` than using a literal result element (`<x:var>`)? Literal result elements are [affected](https://www.w3.org/TR/xslt-30/#lre-namespaces) by the namespaces in the stylesheet.
Actually, it seems `x:var` elements generated by the current code always contain unrelated `xsl` namespace, although right now it doesn't seem harmful.

Reply:
I switched to `xsl:element` here and in two other spots, in a9400dc1e978b44a0144e08b43b2b76406d6490b.